Yoast/wordpress-seo
# Allow filtering of readability score

## Summary

This PR can be summarized in the following changelog entry:

* Adds the possibility to filter posts by readability score.
* Introduces refactored code and more tests than what was previously available.
* Changes the label of the "Bad" score to "Needs improvement".

## Test instructions

This PR can be tested by following these steps:

* Go into the Posts overview
* Ensure you have plenty of posts (3 - 5 should suffice) with varying readability scores and SEO scores.
* Apply a Readability filter based on the Readability score of one (or more) of your posts.
* Also apply SEO filters and check that the specific combination of filters also works.


Fixes #6735 
Fixes #7653
